src/Common/src/System/Security/Cryptography/DerSequenceReader.cs

```diff
@@ -260,16 +284,28 @@ internal string ReadIA5String()
 
         internal DateTime ReadUtcTime()
         {
-            EatTag(DerTag.UTCTime);
+            return ReadTime(DerTag.UTCTime, "yyMMddHHmmss'Z'");
+        }
+
+        internal DateTime ReadGeneralizedTime()
+        {
+            // Currently only supports reading times with no fractional seconds or time differentials
+            // as RFC 2630 doesn't allow these. In case this is done, the format string has to be parsed
+            // to follow rules on X.680 and X.690.
+            return ReadTime(DerTag.GeneralizedTime, "yyyyMMddHHmmss'Z'");
+        }
+
+        private DateTime ReadTime(DerTag timeTag, string formatString)
+        {
+            EatTag(timeTag);
             int contentLength = EatLength();
 
             string decodedTime = System.Text.Encoding.ASCII.GetString(_data, _position, contentLength);
             _position += contentLength;
+            
+            Debug.Assert(decodedTime[decodedTime.Length-1] == 'Z', "The date doesn't follow the X.690 format, doesn't end with 'Z'");
```

Comment:
In these situations, it can be helpful to include in the assert message what data was found that wasn't expected, e.g.
```
$"The date doesn't follow the x.690 format, ending with '{decodedTime[decodedTime.Length-1]}' instead of 'Z'."
```